### PR TITLE
Implemented backoff for transport reconnection

### LINKED ISSUE #68: Implement backoff for transport reconnection
**Feature description**

Currently, when one edge of a transport goes down, the other edge will try to re-establish the transport every three seconds. We should implement a backoff for re-establishing the transport. @evanlinjin suggested using the retry logic in /internal/netutil . Also, we need to improve logging, so that these messages do not clutter stdout:

```[2019-11-13T10:59:10+08:00] WARN [tp:03c775]: failed to redial underlying connection: dial tcp 192.168.0.107:7777: connect: connection refused```

- [x] improve logging
- [x] implement retrial backoff

### PR DESCRIPTION
**Pull request by Kifen**
Did you run `make format && make check`?
yes
Fixes #68  	

 Changes:	
-	updated transport connection redial with a backoff implementation

How to test this PR:
Establish a connection between two visors, then shut down one visor

### CODE REVIEW

File: pkg/router/router_test.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/SkycoinProject/skywire-mainnet/internal/netutil"
```
Comment (nkryuchkov) [inline pkg/router/router_test.go]: Could you please move this line and line 15 below to corresponding import blocks?

File: pkg/transport/managed_transport.go [hunk 2]
```diff
@@ -248,6 +264,17 @@ func (mt *ManagedTransport) dial(ctx context.Context) error {
 	return mt.setIfConnNil(ctx, tp)
 }
 
+// redialConn redials an underlying connection using an exponetial backoff
```
Comment (nkryuchkov) [inline pkg/transport/managed_transport.go]: `exponetial` => `exponential`

File: pkg/transport/manager_test.go [hunk 3]
```diff
@@ -9,6 +9,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/SkycoinProject/skywire-mainnet/internal/netutil"
```
Comment (nkryuchkov) [inline pkg/transport/manager_test.go:12]: Could you please move 12 and 14 to corresponding blocks?

File: pkg/visor/visor_test.go [hunk 4]
```diff
@@ -9,6 +9,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/SkycoinProject/skywire-mainnet/internal/netutil"
```
Comment (nkryuchkov) [inline pkg/visor/visor_test.go]: Could you please move it to corresponding block?

File: pkg/transport/managed_transport.go [hunk 5]
```diff
@@ -151,11 +164,14 @@ func (mt *ManagedTransport) Serve(readCh chan<- routing.Packet, done <-chan stru
 				// If there has not been any activity, ensure underlying 'write' tp is still up.
 				mt.connMx.Lock()
 				if mt.conn == nil {
```
Comment (nkryuchkov) [inline pkg/transport/managed_transport.go]: I would write `if mt.conn == nil && !mt.retrier.alive` to descrease indentation

**Review by nkryuchkov — CHANGES_REQUESTED** [review]

File: pkg/router/router_test.go (hunk 1 above)
Comment (Kifen) [reply pkg/router/router_test.go]: done

File: pkg/transport/managed_transport.go (hunk 2 above)
Comment (Kifen) [reply pkg/transport/managed_transport.go]: corrected

File: pkg/transport/manager_test.go (hunk 3 above)
Comment (Kifen) [reply pkg/transport/manager_test.go:12]: done

File: pkg/visor/visor_test.go (hunk 4 above)
Comment (Kifen) [reply pkg/visor/visor_test.go]: done

**Review by nkryuchkov — CHANGES_REQUESTED** [review]
Great work! Please just remove the committed binaries

**Review by nkryuchkov — APPROVED** [review]
Actually, I haven't seen that the binaries are in the `vendor` folder, so that's OK. Maybe just running `make dep` would be great.

**Comment by jdknives** [comment]
Sorry again @Kifen . But the issue tackled with this PR will be fixed in #166

**Pull request closed without merging**